src/base/static/libs/maps/mapboxgl-provider.js

```diff
@@ -433,21 +433,21 @@ export default (container, options) => {
     ];
   };
 
-  const createVectorTileLayer = async ({
-    id,
-    url,
-    style_url,
-    source_layer,
-  }) => {
+  const createVectorTileLayer = async ({ id, url, layers, source_layer }) => {
     sourcesCache[id] = {
       type: "vector",
       tiles: [url],
     };
     map.addSource(id, sourcesCache[id]);
 
-    const style = await VectorTileClient.fetchStyle(style_url);
+    // If layers is an array, assume we have a config-based set of mapbox
+    // layers. Otherwise, assume we have a url to a remote-hosted mapbox
+    // stylesheet.
+    layers = Array.isArray(layers)
+      ? layers
+      : await VectorTileClient.fetchLayers(layers);
```

Comment:
Calling this `layers` is confusing, because we already have a `layers` array in our config, and now we are adding a `layers` field to each of those `layers`, which gives us `layer.layers`. I'm not sure what to call them, but `layers.layers` might cause confusion. Is there a better word we can use here? Perhaps `mapbox_layers`?

It also seems strange to overload our new `layers` field to handle both url's and an array of "layer" objects. When we move to a db storage, we can't model our data like this. I think we should maintain separate fields here, like `style_url`, which is a nullable string, and `layers`, which is a nullable array of "mapbox layers".

Reply:
Yeah, I was wondering about that too. Let's just have a separate `style_url` and `mapbox_layers` entries in the config, which is a pattern we can repeat for geojson layers down the road.